target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network

```diff
@@ -29,9 +32,30 @@ ipq60xx_setup_interfaces()
 	esac
 }
 
+ipq60xx_setup_macs()
+{
+	local board="$1"
+	local lan_mac=""
+	local wan_mac=""
+	local label_mac=""
+
+	case "$board" in
+	qihoo,360v6)
+		lan_mac=$(mtd_get_mac_ascii factory lanMac)
+		wan_mac=$(macaddr_add "$lan_mac" 1)
+		label_mac=$lan_mac
+		;;
```

Comment:
This should be replaced via NVMEM which now can read ASCII MAC-s as well

Reply:
It doesnt have to be, you can use NVMEM on SMEM partition tables.

Reply:
Do you mean something like this?
```
	nand@0 {
		reg = <0>;
		nand-ecc-strength = <4>;
		nand-ecc-step-size = <512>;
		nand-bus-width = <8>;

		partitions {
			compatible = "qcom,smem-part";

			partition-0-factory {
				compatible = "fixed-partitions";
				label = "factory";
				read-only;
				#address-cells = <1>;
				#size-cells = <1>;

				partition@0 {
					reg = <0x0 0x40000>;

					macaddr_factory_ethaddr: lanMac {};
				};
			};
		};
	};
```

Reply:
Close, but you cannot use fixed partitions on it:
https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/Documentation/devicetree/bindings/mtd/partitions/qcom,smem-part.yaml?h=next-20241128#n45

Reply:
Unfortunately NVMEM on SMEM partition table does not work:
```
	nand@0 {
		reg = <0>;
		nand-ecc-strength = <4>;
		nand-ecc-step-size = <512>;
		nand-bus-width = <8>;

		partitions {
			compatible = "qcom,smem-part";
			#address-cells = <1>;
			#size-cells = <1>;

			partition-factory {
				compatible = "nvmem-cells";
				#address-cells = <1>;
				#size-cells = <1>;
				label = "factory";
				read-only;

				macaddr_lan: macaddr@47 {
					compatible = "mac-base";
					reg = <0x47 0x11>;
					#nvmem-cell-cells = <1>;
				};
			};
		};
	};
```

Reply:
Ahh, its u-boot-env variable, yeah in that case only ethaddr, eth1addr and so on are parseable for MAC-s

Reply:
https://github.com/openwrt/openwrt/blob/39b0f081ecaf4fb9acc622d694dc52293a4dda1f/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch#L170-L174

It's actually just "ethaddr", so can we keep using the script?

Reply:
Yes, untill U-Boot driver gets converted to layouts